apps/passport-client/src/worker/service-worker.ts

```diff
@@ -53,6 +59,18 @@ const EPHEMERAL_CACHE_RESOURCES = new Set([
   ...GENERATED_CHUNKS
 ]);
 
+/**
+ * Z cache is for Z API-relevant resources which do not change frequently, such
+ * as GPC artifacts, where we always use the cached version if available.  This
+ * cache is only updated if the artifact to be fetched is not in the
+ * cache. Previously cached versions are cleared for space-saving purposes.
+ *
+ * Cache name includes deployed artifact version, which protects against using
+ * cached artifacts of a different version.
+ */
+const Z_CACHE_NAME = `${CACHE_VERSION}-${GPC_ARTIFACTS_NPM_VERSION}-Z`;
```

Comment:
Something I realized while reviewing your next PR extending the family.
If we want to expand the family by adding circuits without breaking existing ones, we can do so safely and express it with an NPM patch update (0.5.0 -> 0.5.1).  Semantic versioning will say the right thing about compatibility there.
The use of the version here is stricter than that, because it will clear the cache on any version change, not just an incompatible one.
That makes me think maybe it would b emore right to have a separate constant for the compatible version (major.minor) to use here, so people's caches don't get cleared when we expand the family.
That being said, it adds complexity, and is something we could get wrong if we screw it up, so I'm not sure whether to recommend it or not.
Think about it.  It's at least not something which can't be changed later, since if we change the way the cache name is calculated, it'll result in clearing the cache anyway, so it'll fail in the direction of correctness via more network fetches.